src/service/video-manager-impl.js

```diff
@@ -728,15 +728,17 @@ class VideoEntry {
         adEnd.bind(this)));
 
     function onInteraction() {
+      const {video} = this;
       this.firstPlayEventOrNoop_();
       this.userInteractedWithAutoPlay_ = true;
-      this.video.showControls();
-      this.video.unmute();
+      video.showControls();
+      video.unmute();
       unlisteners.forEach(unlistener => {
         unlistener();
       });
       removeElement(animation);
       removeElement(mask);
+      video.signals().signal(VideoServiceSignals.USER_INTERACTED);
```

Comment:
I think this method needs to listen to `USER_INTERACTED` instead of firing it (and having `click` handlers above fire it).

This event will be helpful for https://github.com/ampproject/amphtml/issues/9055 where we want AMP actions such as `play()`, `pause()` to be considered `user-interactions` which then we can easily do by triggering this event ( so maybe move the event definition to where other events like PLAYING, etc.. are in `video-interface`?)

Reply:
Welp, the signal is permanent, so the state would be the same regardless as to where the actual event is handled. I propose removing `userInteractedWithAutoplay` in favor of the signal (`signals.get(USER_INTERACTED)`), which should reconcile #9055 (in a follow-up).

Reply:
Yup, on the same page!